File: VB/RegistrationForm.Lesson4/DataModel/Gender.vb
Imports System.Collections.Generic

Namespace RegistrationForm.DataModel
    Public Class Gender
        Private privateID As Integer
        Public Property ID() As Integer
            Get
                Return privateID
            End Get
            Private Set(ByVal value As Integer)
                privateID = value
            End Set
        End Property
        Private privateDescription As String
        Public Property Description() As String
            Get
                Return privateDescription
            End Get
            Private Set(ByVal value As String)
                privateDescription = value
            End Set
        End Property
        Public Sub New(ByVal id As Integer, ByVal description As String)
            Me.ID = id
            Me.Description = description
        End Sub
    End Class
    Public Class GenderList
        Inherits List(Of Gender)

        Private Shared privateSource As GenderList
        Public Shared Property Source() As GenderList
            Get
                Return privateSource
            End Get
            Private Set(ByVal value As GenderList)
                privateSource = value
            End Set
        End Property

        Shared Sub New()
            Source = New GenderList()
            Source.Add(New Gender(0, "Female"))
            Source.Add(New Gender(1, "Male"))
        End Sub
    End Class
End Namespace

File: CS/RegistrationForm.Lesson1/DataModel/Gender.cs
using System.Collections.Generic;

namespace RegistrationForm.DataModel {
    public class Gender {
        public int ID { get; private set; }
        public string Description { get; private set; }
        public Gender(int id, string description) {
            ID = id;
            Description = description;
        }
    }
    public class GenderList : List<Gender> {
        public static GenderList Source { get; private set; }

        static GenderList() {
            Source = new GenderList();
            Source.Add(new Gender(0, "Female"));
            Source.Add(new Gender(1, "Male"));
        }
    }
}
